deep_qa/layers/backend/repeat_like.py

```diff
@@ -0,0 +1,63 @@
+from keras import backend as K
+from overrides import overrides
+
+from ..masked_layer import MaskedLayer
+
+
+class RepeatLike(MaskedLayer):
+    """
+    This ``Layer`` is like :class:`~.repeat.Repeat`, but gets the number of repetitions to use from
+    a second input tensor.  This allows doing a number of repetitions that is unknown at graph
+    compilation time, and is necessary when the ``repetitions`` argument to ``Repeat`` would be
+    ``None``.
+
+    If the mask is not ``None``, we must be able to call ``K.expand_dims`` using the same axis
+    parameter as we do for the input.
+
+    Input:
+        - A tensor of arbitrary shape, which we will expand and tile.
+        - A second tensor whose shape along one dimension we will copy
+
+    Output:
+        - The input tensor repeated along one of the dimensions.
+
+    Parameters
+    ----------
+    axis: int
+        We will add a dimension to the input tensor at this axis.
+    copy_from_axis: int
+        We will copy the dimension from the second tensor at this axis.
+    """
+    def __init__(self, axis: int, copy_from_axis: int, **kwargs):
```

Comment:
I guess I was thinking that the tensor you want to tile to would be passed to the init method of `Repeat`, but go for this way, sure.